### PR TITLE
Convert nullish values to empty strings.

### PR DESCRIPTION
**Pull request by bigeasy**
Convert nullish values to empty strings `_serializeValue`. See #98.

### CODE REVIEW

**Comment by juliangruber** [comment]
LGTM, can you add a test though so we make sure this change stays?

**Comment by bigeasy** [comment]
@juliangruber Actually, there is a test in the `abstract/put-get-del-test.js` file, but it is in the `nonErrorValues` function and the call to that function in `test.js` is commented out.

Should I add a collection of serialization tests in the "extensibility" section?

**Comment by juliangruber** [comment]
hm we'd rather figure out why the test is commented out in the first place

**Comment by bigeasy** [comment]
I would assume because it would require a concrete implementation to run those particular tests. None of the abstract tests that actually do anything against a concrete implementation are run in `test.js`. For example, the `open` test of `abstract/open-test.js` is not called. In fact, only the `args` test which tests the argument validation to the open function is called.

This is why I thought you might want to test just the serialization specifically, as a unit. It appears as though your counting on down streams to test an integration.

**Comment by juliangruber** [comment]
ok yeah, that's very good reasoning, @bigeasy, I fully agree!

**Comment by bigeasy** [comment]
`tape` documentation says that `t.same` is an alias for `t.deepEqual`. That's interesting. I'd expect it to be `===` strict quality. Didn't experiment.

Note that I set `process.browser = true` and then delete it. Let me know if I should preserve and reset. Do you actually run these tests in the browser or ever hope to?

**Comment by juliangruber** [comment]
"equal" means it's the exactly equal thing, for example there can't be two equal bottle caps as each will be slightly different.

"same" means it's of the same kind, so for example two bottle caps of the same making are considered the same.

This is why `t.same` is an alias for `t.deepEqual`, as it's checking for same value and _not_ reference equality (=identity).

File: test.js [hunk 1]
```diff
@@ -62,6 +62,30 @@ test('test core extensibility', function (t) {
...
+  function Test (location) {
+    AbstractLevelDOWN.call(this, location)
+  }
+
+  util.inherits(Test, AbstractLevelDOWN)
+
+  var buffer = new Buffer(0)
+  var test = new Test('foobar')
+
+  t.ok(test._serializeKey(1) === '1', '_serializeKey converts to string')
```
Comment (juliangruber) [inline test.js]: this should be written using `t.equal()`, so we get more useful error messages in case it doesn't match

File: test.js [hunk 2]
```diff
@@ -62,6 +62,30 @@ test('test core extensibility', function (t) {
...
+  t.ok(test._serializeKey(buffer) === buffer, '_serializeKey returns Buffer as is')
+
+  t.ok(test._serializeValue(1) === '1', '_serializeValue converts to string')
+  t.equal(test._serializeValue(null), '', '_serializeValue converts null to empty string')
+  t.ok(test._serializeValue(buffer) === buffer, '_serializeValue returns Buffer as is')
+
+  process.browser = true
+  t.ok(test._serializeValue(1) === 1, '_serializeValue returns value as is when process.browser')
+  delete process.browser
```
Comment (juliangruber) [inline test.js]: this test suite might also be run inside the browser, so we have a problem here. i'd say the straight forward solution is to have `if (process.browser) { ... } else { ... }` inside the tests.

File: test.js (hunk 1 above)
Comment (bigeasy) [inline test.js]: We're testing to see if converts to string, hence the strict equality, but I'll rewrite using `t.equal`.

Comment (bigeasy) [inline test.js]: Oh, now I see that `t.equal` is based on `===`.

**Comment by bigeasy** [comment]
Getting closer, I think. Your thoughts @juliangruber?

**Comment by juliangruber** [comment]
i think this looks good, but also i feel like i'm missing some background / prior discussion about null values. what do other maintainers think?

**Comment by ralphtheninja** [comment]
@bigeasy @juliangruber I've rebased this onto latest master. Made a new PR https://github.com/Level/abstract-leveldown/pull/119

**Pull request closed without merging**